ElementX/Sources/Services/Authentication/AuthenticationCoordinator.swift

```diff
@@ -123,58 +120,48 @@ class AuthenticationCoordinator: Coordinator {
         coordinator.start()
     }
     
-    private func login(username: String, password: String, completion: @escaping (Result<Void, AuthenticationCoordinatorError>) -> Void) {
+    private func login(username: String, password: String) async -> Result<Void, AuthenticationCoordinatorError> {
         Benchmark.startTrackingForIdentifier("Login", message: "Started new login")
         
         delegate?.authenticationCoordinatorDidStartLoading(self)
         
-        DispatchQueue.global(qos: .background).async { [weak self] in
-            guard let self = self else { return }
-            
-            do {
-                self.setupUserSessionForClient(try loginNewClient(basePath: self.baseDirectoryPathForUsername(username),
-                                                                  username: username,
-                                                                  password: password))
-                
-                DispatchQueue.main.async {
-                    completion(.success(()))
-                }
-            } catch {
-                MXLog.error("Failed logging in with error: \(error)")
-                
-                DispatchQueue.main.async {
-                    completion(.failure(.failedLoggingIn))
-                }
-            }
+        let basePath = baseDirectoryPathForUsername(username)
+        async let loginTask = Task.detached {
+            try loginNewClient(basePath: basePath,
+                               username: username,
+                               password: password)
+        }
+        
+        switch await loginTask.result {
+        case .success(let client):
+            return await setupClientProxyForClient(client)
+        case .failure(let error):
+            MXLog.error("Failed logging in with error: \(error)")
+            return .failure(.failedLoggingIn)
         }
     }
     
-    private func restorePreviousLogin(_ usernameTokenTuple: (username: String, token: String), completion: @escaping (Result<Void, AuthenticationCoordinatorError>) -> Void) {
+    private func restorePreviousLogin(_ usernameTokenTuple: (username: String, token: String)) async -> Result<Void, AuthenticationCoordinatorError> {
```

Comment:
Nit-pick but could this be `accessToken` just to be clear that its different from a `token` based login.
```suggestion
    private func restorePreviousLogin(_ usernameTokenTuple: (username: String, accessToken: String)) async -> Result<Void, AuthenticationCoordinatorError> {
```